pkg/collector/corechecks/servicediscovery/model/model.go

```diff
@@ -0,0 +1,32 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package model contains types for service discovery.
+package model
+
+// Port represents an open port in the current host.
+type Port struct {
+	PID         int    `json:"pid"`
+	ProcessName string `json:"processName"`
+	Port        int    `json:"port"`
+	Proto       string `json:"proto"`
+}
+
+// OpenPortsResponse is the response for the system-probe /service_discovery/open_ports endpoint.
+type OpenPortsResponse struct {
+	Ports []*Port `json:"ports"`
+}
+
+// Proc represents a system process.
+type Proc struct {
+	PID     int      `json:"pid"`
+	Environ []string `json:"environ"`
+	CWD     string   `json:"cwd"`
+}
+
+// GetProcResponse is the response for the system-probe /service_discovery/procs/{pid} endpoint.
+type GetProcResponse struct {
+	Proc *Proc `json:"proc"`
+}
```

Comment:
What is the benefit of having 2 endpoints (`open_ports` and `/procs/{pid}`) as opposed a single endpoint that returns:
- `PID`, `ProcessName`, `Port`, `Proto`, `Environ`, and `CWD`

Hosting 2 endpoints in the system probe, querying them from core agent, and then needing to combine their responses seems more complex than having a single endpoint that returns everything needed.